### PR TITLE
feat(plc): drag-resize grid + live tile bodies + cross-PLC analytics + shared library primitives + members invite (Phases 1–4, 6)

### PR DESCRIPTION
**Pull request by OPS-PIvers**
## Summary

Multi-phase overhaul of the PLC dashboard addressing the user's pain points: rigid bento containers, tiles that are "glorified links to tabs", weak shared assignment data, and shallow PLC analytics. Six phases shipped on this branch.

### Phase 1 — Drag-resizable v2 grid (commits `bd557c7`, `cfa6aa7`)
- New 12-column free-form grid with `{x, y, w, h}` coords per tile (`components/plc/grid/`)
- Drag-to-resize from 8 edge/corner handles via Pointer Events + rAF (modeled on `DraggableWindow`)
- Drag-to-reorder via grip → `commitTileCoords` swaps positions and push-down repacks neighbors
- Tile content stays interactive in edit mode (drag/resize scoped to chrome handles, no `pointer-events-none`)
- Backward-compatible layout migration: legacy `size`-only docs migrate to `coords` on read; both fields persist
- Behind `localStorage.spart.plcDashboard.gridV2` opt-in flag
- Two parallel review agents surfaced 5 issues; all addressed in `cfa6aa7` (no-op reorder, corner/content collisions, listener leak, parseTile fallback, hint copy)

### Phase 2A — Live tile bodies + fullscreen expansion (commit `9a21927`)
- `NotesBody` and `TodosBody` extracted from `Plc{Notes,Todos}Tab` into `components/plc/bodies/`
- Tabs become 3-line wrappers around the body (single source of truth)
- v2 grid tiles expose an expand button on hover (top-right) when their kind has a real fullscreen body
- `PlcDashboard` owns `expandedTile` state; renders the body in a fullscreen overlay with back button
- Escape collapses fullscreen before falling back to the dashboard's own close behavior
- Quiz/Assignments body extraction deferred (1500 lines mechanical; fullscreen overlay already renders existing tabs cleanly)

### Phase 3 — Cross-PLC analytics body + tile (commit `7768207`)
- `PlcAnalyticsBody` reads `usePlcContributions(plcId)` and groups by `syncGroupId ?? quizId` so contributions from the same logical quiz aggregate across all teachers who ran it
- Quiz titles joined from `usePlcAssignments` templates with question-text fingerprint fallback
- Two modes: compact (top-N quiz cards in tile) + full (per-quiz drilldown with `PlcAggregateSection`'s score distribution + per-question accuracy + schema-drift banner)
- `CompletedAssignmentsTile` rewritten to render compact analytics; legacy "View all" footer button gone
- Refactor: `aggregateGroup`, `groupBySchema`, `SchemaGroup`, `PlcAggregate`, `SCORE_BUCKETS` hoisted from `PlcTab.tsx` into pure `plcAnalyticsAggregate.ts`
- Cross-PLC analytics replace the Google Sheet link as the primary surface for "how did the PLC do?"

### Phase 4 — Shared library primitives (commit `dc5109a`)
- `LibraryPreviewPane` — right-side detail pane (header + body + footer with primary/secondary actions). Esc closes; backdrop click does NOT (keeps grid interactive). Scoped to container query, not portalled.
- `BulkActionBar.actions: BulkAction[]` — generalized from fixed Move/Delete to arbitrary action list (duplicate, archive, share). Legacy `folders/onMove/onDelete` props remain for incremental migration.
- `libraryDuplicate.buildDuplicateAction()` + `suggestDuplicateTitle()` — kebab-menu helper + idempotent " (Copy N)" title bumper.
- `LibraryShell.visibleTabs?: LibraryTab[]` — embedded surfaces (PLC dashboard tile) can hide In Progress/Archive.
- No manager changes yet — Phase 5 (per-widget adoption) deferred; primitives are ready for incremental wiring.

### Phase 6 — MembersBody + default layout coords (commit `1562248`)
- `MembersBody` (compact + full modes): inline invite form (lead-only) with email validation + send via `usePlcInvitations`, pending-invitations list with revoke, per-member remove (lead-only) for non-lead non-self members. Compact mode disables the invitation listener so the tile preview doesn't add Firestore reads.
- `MembersTile` rewritten as a thin wrapper around `MembersBody compact`. `members` added to `EXPANDABLE_KINDS`; fullscreen overlay shows the full management UI.
- `DEFAULT_PLC_OVERVIEW_LAYOUT` now stamps explicit `coords` on every default tile — no more reliance on the migrator for first-time users. Each tile keeps its legacy `size` alongside so the v1 bento grid still works.

## Test plan

- [x] `pnpm run validate` clean (typecheck + lint + format + 2319 tests + 204 functions tests)
- [x] 18 new migration unit tests pass
- [x] Existing layout-hook test unaffected (legacy `size`-only docs parse and write correctly)
- [ ] Manual: opt into `localStorage.spart.plcDashboard.gridV2 = 'true'`, open PLC dashboard
- [ ] Manual: enter edit mode → drag grip → reorder; drag corner → resize with neighbors push-down
- [ ] Manual: hover any expandable tile → click expand → fullscreen body renders; Esc returns
- [ ] Manual: notes inline editor saves; reload preserves
- [ ] Manual: todos toggle/add/edit inline + in fullscreen
- [ ] Manual: completedAssignments fullscreen → click a quiz card → drilldown shows score buckets + per-question accuracy
- [ ] Manual: members fullscreen → enter email → send invite (lead-only); revoke pending invite
- [ ] Manual: members fullscreen → remove a member (lead-only, non-self)
- [ ] Manual: rules emulator confirms `coords` accepted, missing `coords` still accepted (back-compat)
- [ ] Manual: mobile (<768px) collapses to single column, resize hidden, reorder via long-press still works

## Deferred to follow-up PRs
- **Phase 2B**: extract `PlcQuizLibraryBody`, `PlcAssignmentsBody` (1500-line mechanical refactor — currently fullscreen overlay just renders the existing tab components, which works fine)
- **Phase 5**: Quiz/VA/MiniApp/GL managers adopt Phase 4 primitives (preview pane + duplicate + multi-action bulk toolbar; requires per-widget duplicate plumbing in each hook)
- **Retire `gridV2` flag default**: flip default to v2 once PLC dogfooding confirms stability
- **Mobile edit mode**: long-press resize / handle redesign for touch (currently desktop-only via viewport check)
- **Resize keyboard accessibility**: Shift+Arrow on focused grip

https://claude.ai/code/session_01D6Ex4ZBBHVPAveDmzUZSxY

### CODE REVIEW

File: components/plc/grid/PlcGridLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    };
+    recompute();
+    const ro = new ResizeObserver(recompute);
+    ro.observe(el);
+    window.addEventListener('resize', recompute);
+    return () => {
+      ro.disconnect();
+      window.removeEventListener('resize', recompute);
+    };
+  }, []);
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/plc/grid/PlcGridLayout.tsx:143]: No automated fix — this is a performance optimization tradeoff, not a bug, and switching from time-based debounce to `requestAnimationFrame` for the `window` resize listener requires manual benchmarking against the `ResizeObserver` callback (which already runs at frame cadence on most engines) to confirm it actually wins. Leaving this for human review as an architectural decision; happy to revisit if the team wants the rAF pattern adopted across the grid layout module.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/plc/grid/PlcGridLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+ *
+ * Layout state is owned by the caller (`PlcOverviewTab` via
+ * `usePlcOverviewLayout`); this component is a controlled view.
+ */
+export const PlcGridLayout: React.FC<PlcGridLayoutProps> = ({
+  plc,
+  layout,
+  editMode,
+  onLayoutChange,
+  onNavigateTab,
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/plc/grid/PlcGridLayout.tsx:100]: No automated fix — this requires product context. The reviewer correctly identifies that the PR description claims "mobile reorder via long-press still works" but the implementation passes `editMode={editMode && !isMobile}` to `PlcGridTile`, which disables `useSortable`. The correct resolution depends on product intent (should mobile have reorder-without-resize, or is the description wrong?), so flagging for human decision rather than picking one arbitrarily.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/common/library/LibraryPreviewPane.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+/**
+ * Right-side detail pane for library managers. Renders inside the manager's
+ * tab content area as a sibling of the grid, not as a portal — that keeps
+ * the pane scoped to the widget's container query so a small widget gets
+ * a small pane and a fullscreen widget gets a large one.
+ *
+ * The pane animates in from the right; closing animates back out (handled
+ * by the absent-from-DOM pattern, so React can reuse the slide-in
+ * `animate-in` classes on every open). Esc closes; backdrop click does
+ * NOT (this isn't a modal — the underlying grid stays interactive).
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/common/library/LibraryPreviewPane.tsx]: Fixed in 20637623 — reworded the docstring to accurately describe the unmount-on-close behavior (no exit animation; the slide-in `animate-in` classes are re-applied on the next open).

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by OPS-PIvers — COMMENTED** [review]
## Automated Code Review

**Date:** 2026-05-12

### Summary
Six-phase overhaul of the PLC dashboard: drag-resizable v2 grid (behind `localStorage.spart.plcDashboard.gridV2` opt-in), live-content tile bodies with fullscreen expansion, cross-PLC analytics aggregation, shared library primitives (`LibraryPreviewPane`, `BulkActionBar.actions`, `libraryDuplicate`), and a new `MembersBody` with inline invite/revoke. Touches 28 files (+3,053/-1,001), introduces 4 new bodies, 4 new grid modules, 3 library primitives, and a 292-line migration test suite.

### Code Quality
- The Notes/Todos/Members extraction is sound — the tab files shrink dramatically (`PlcNotesTab.tsx` -378, `PlcTodosTab.tsx` -217, `MembersTile.tsx` -66) while the bodies become reusable in both the tab and the v2 grid fullscreen overlay. Single-source-of-truth pattern is clean.
- `plcAnalyticsAggregate.ts` (177 lines) was hoisted out of `PlcTab.tsx` as pure functions — good for testability.
- The two outstanding review-bot comments are now both replied to (rAF perf is a tradeoff, not a bug; mobile-reorder mismatch requires product input).
- Docstring on `LibraryPreviewPane` was clarified in the latest commit.

### Widget Standards
✓ Not applicable — this PR is PLC dashboard surface, not a widget. No `components/widgets/` files touched.

### Admin Config Standards
✓ Not applicable.

### Firestore & Security
- No Firestore rules changes in the diff. The `coords` field is additive on `Plc.overviewLayout.tiles[]`, and the PR description references a rules-emulator manual-test step confirming `coords` is accepted alongside legacy `size`. Worth confirming the rules actually allow the new field shape before merging — flagging as a manual verification gate.
- `usePlcInvitations` and `usePlcContributions` are existing hooks; no new collections appear in the file list.

### Regression Risk
- **Medium overall, gated by feature flag.** v2 grid is opt-in via `localStorage`; default render path stays on v1 `PlcBentoGrid`. The migrator (`deriveCoordsFromLegacy` in `tileGridMath.ts`) is back-compat — legacy `size`-only docs stamp `coords` on first read, both fields persist.
- **types.ts +86/-14** — adds `PlcGridCoords`, expands `PlcBentoTile`. Verify the union didn't change in a way that breaks existing serialized docs in production Firestore.
- **PlcDashboard.tsx +144/-39** — owns the new `expandedTile` state and fullscreen overlay. Esc-handling is layered; confirm it doesn't intercept Esc from focused inputs inside fullscreen bodies (e.g., notes editor) and accidentally close the overlay mid-edit.
- **CompletedAssignmentsTile.tsx -83 lines** — entire "View all" footer button removed. If any analytics/onboarding relies on that interaction, it's gone now.
- Phase 2B (Quiz/Assignments body extraction) is explicitly deferred — fullscreen overlay falls back to rendering existing tab components, which the PR description notes works fine. Acceptable.

### Missing Tests
- ✓ `plcLayoutMigration.test.ts` (292 lines) covers the migrator.
- Gap: no test for `commitTileCoords` push-down repacking on resize, the cross-PLC analytics grouping by `syncGroupId ?? quizId`, or the `MembersBody` lead-only gating logic (invite/revoke/remove). These are the higher-risk new behaviors. Worth adding unit tests for `commitTileCoords` at minimum before flipping the flag default.

### Overall
**Merge readiness:** Ready with minor notes
PR is well-scoped per phase, feature-flag-gated, and has solid migration test coverage. Manual test plan is appropriate — recommend completing the rules-emulator and mobile-edit-mode items before merge, and adding `commitTileCoords` unit tests as a follow-up.

---
_This review was generated automatically. All findings should be verified by a human before merging._

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/plc/grid/PlcGridLayout.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/plc/grid/PlcGridLayout.tsx:100]: Fixed in 9cdf5618 — the PR description was right and the implementation was wrong. Split the prop: `editMode` now flows through unchanged so `useSortable` activates on mobile (long-press reorder works), and a new `showResizeHandles` prop is gated on `!isMobile` to suppress only the 6px edge handles (which aren't touch-friendly). Mobile users now keep the long-press reorder the description promised. Resize stays desktop-only by design — Phase 6 follow-up could add a touch-friendly resize gesture if needed.

---
_Generated by [Claude Code](https://claude.ai/code)_
